### ISSUE TITLE
Proposal: Introduce ErrorBoundary as must-have

### ISSUE THREAD

**Issue by usrbowe** [facebook/react-native-website#1042]
I would like to add `ErrorBoundary` introduction into RN documentation. 
And highlight how crucial this is to be implemented. 
It makes the app much less prone to fatal crashes, just because of some `name of undefined` errors.
I know it is well described in React documentation, but for RN purpose I would focus on explaining how easily app can crash and add ErrorBoundary as one of the ways, to prevent fatal crash of app.
Ideally with simple example, showing nested ErrorBoundary, from which user can recover (ie. going back to prev screen).

**Comment by rickhanlonii** [comment]
Hey @usrbowe, this is a great idea! Where do you think it would go? We should be sure that this is RN specific and not just re-stating what's in the React docs. It would also be great to explain how error boundaries interact with reboxes

**Comment by usrbowe** [comment]
Hi @rickhanlonii,

I think it can be either under [Improve UX](https://facebook.github.io/react-native/docs/improvingux) or create a new section under **Guides**.

I also thought about the fact, that the documentation already exist in React docs, but I would like to highlight importance of having this component at least in root of app, to prevent app crashes, which can be prevent, when using this component.

Another point for putting this directly in RN, I can't seem to find many open sourced popular RN apps, which would have this. So it indicates, there might be awareness problem. Same goes with articles on this topic, which I can't seem many specific for RN.

I believe this can improve UX a lot, if we can catch the error and provide nice fallback component.

I didn't get: 
>reboxes
Do you mean red boxes in dev environment? If so, yeah, I saw one issue, which confused devs a lot, when they implemented boundary, but still saw red boxes, so they taught it's not working.

I just believe this small change, adding one component, should be almost `mandator` to all apps, since it such a game changer. Instead of having app completely crashed, it can be catched and app can even sometimes recover (even with JS reload feature)

**Comment by rickhanlonii** [comment]
I think under Improve UX makes sense! I would push back on creating a new guide since we're in the process of reviewing the guides and docs we have now to clean them up

I agree this would improve UX, there should be at least a root error boundary. One thing that may help is if the starter template included an error boundary so it exists out of the box

>when they implemented boundary, but still saw red boxes, so they taught it's not working

Yeah this is what I mean!

**Comment by usrbowe** [comment]
Yep, it can be just a mention with a link to React docs for more details. 
And if it can be included in default template and one paragraph explaining its purpose, with highlighted issue when running on DEV. 
I think this would be sufficient 👍
